dds_web/api/errors.py

```diff
@@ -68,23 +105,16 @@ def __init__(
         username=None,
         message="The user does not have the necessary permissions.",
     ):
-        super().__init__(message)
-
-        if not username:
-            username = auth.current_user()
+        if username:
+            structlog.threadlocal.bind_threadlocal(user=username)
```

Comment:
Do we need to release after binding too or is this fine? Or unbind or what it's called

Reply:
In theory yes and there is also the  s`tructlog.threadlocal.clear_threadlocal()` command to clear the threadlocal cache. However, I have mostly struggled with a lack of persistence rather than the other way round during development. I will do a bit more testing.

Reply:
I have now included a `structlog.threadlocal.clear_threadlocal()` after the log is called. 

In particular for the exceptions it may be desirable to be able to gradually assign values to the local context, which are added to the log event in case an exception occurs. Therefore, I think it is preferable to not clear the context before the event is logged. 

But once the log event is written, I agree, it is good practise to immediately clear out the context to not have forgotten assignments lingering around and obfuscating subsequent log events.

Reply:
Ah I saw you added it to within the [context manager](https://github.com/ScilifelabDataCentre/dds_web/pull/771/files#diff-bacb9c85e5e141775ed3b7a626bf14730d2ec362398022c166bead23297eecc8R91) -- when the `bound_threadlocal` is used as a context manager that itself should actually clean it up after. So there shouldn't be a need for the clearing within this. I thought it would be in the exception in this case, but if not then we shouldn't need this at all 👍🏻

Reply:
@MatthiasZepper https://www.structlog.org/en/stable/api.html#structlog.threadlocal.bound_threadlocal

Reply:
So we an remove the `structlog.threadlocal.clear_threadlocal()` correct?

Reply:
Depending on the behavior that we want. I initially felt that it might be helpful to explicitly bind variables to the context for a longer time such that they are retained for multiple log entries, but later agreed that the risk of having some forgotten binding around is to high, so I included the clear command.

The source of `bound_threadlocal` is:
```
def bound_threadlocal(**kw: Any) -> Generator[None, None, None]:
    context = get_threadlocal()
    saved = {k: context[k] for k in context.keys() & kw.keys()}

    bind_threadlocal(**kw)
    try:
        yield
    finally:
        unbind_threadlocal(*kw.keys())
        bind_threadlocal(**saved)
```

So it seems to save and restore only values that are explicitly assigned within the `bound_threadlocal`, but not keys that will be assigned within the context later.

```
bind_threadlocal(a=1, b=2, c=3)
get_threadlocal()   # {'a': 1, 'b': 2, 'c': 3}

with  bound_threadlocal(a=4, b=5, d=6):
        bind_threadlocal(c=5, e=7)
        # a and b from the context manager take precedence over the initial bindings
        get_threadlocal()   #{'a': 4, 'b': 5, 'c': 5, 'd': 6, 'e': 7}

get_threadlocal()  # {'a': 1, 'b': 2, 'c': 5, 'e': 7}
# c was not restored, d is not retained

clear_threadlocal()
get_threadlocal() # {}
```

Reply:
Ok! Thanks for the demonstration! Lets keep the clear_threadlocal line.